charts/opentelemetry-kube-stack/values.yaml

```diff
@@ -0,0 +1,335 @@
+  # # Environment variables for the target allocator
+  # env:
+  #   - name: ENV_VAR1
+  #     value: value1
+  #   - name: ENV_VAR2
+  #     value: value2
+  # # Observability configuration for the target allocator
+  # observability:
+  #   metrics:
+  #     enableMetrics: true
+
+  # Affinity configuration for the collector
+  affinity: {}
+  #   nodeAffinity:
+  #     requiredDuringSchedulingIgnoredDuringExecution:
+  #       nodeSelectorTerms:
+  #         - matchExpressions:
+  #             - key: kubernetes.io/e2e-az-name
+  #               operator: In
+  #               values:
+  #                 - e2e-az1
+  #                 - e2e-az2
+
+  # Lifecycle configuration for the collector
+  lifecycle: {}
+  #   preStop:
+  #     exec:
+  #       command:
+  #         [
+  #           "/bin/sh",
+  #           "-c",
+  #           "echo Hello from the preStop handler > /dev/termination-log",
+  #         ]
+
+  # Liveness probe configuration for the collector
+  livenessProbe: {}
+  #   initialDelaySeconds: 3
+  #   periodSeconds: 5
+  #   timeoutSeconds: 2
+  #   failureThreshold: 5
+
+  # Observability configuration for the collector
+  observability: {}
+  #   metrics:
+  #     enableMetrics: true
+
+  # Update strategy for the collector
+  updateStrategy: {}
+  #   type: RollingUpdate
+
+  # Volume mounts for the collector
+  volumeMounts: []
+  #   - name: data
+  #     mountPath: /data
+
+  # Ports configuration for the collector
+  # The operator automatically calculates ports for known receivers and exporters
+  # Set any custom ports here.
+  ports: []
+  # - name: http
+  #   protocol: TCP
+  #   port: 80
+  #   targetPort: 8080
+
+  # Environment variables for the collector
+  env: []
+  # - name: ENV_VAR1
+  #   value: value1
+  # - name: ENV_VAR2
+  #   value: value2
+
+  # Volume claim templates for the collector
+  volumeClaimTemplates: []
+  # - metadata:
+  #     name: storage
+  #   spec:
+  #     accessModes: ["ReadWriteOnce"]
+  #     resources:
+  #       requests:
+  #         storage: 1Gi
+
+  # Tolerations for the collector
+  tolerations: []
+  # - key: "key"
+  #   operator: "Equal"
+  #   value: "value"
+  #   effect: "NoSchedule"
+
+  # Volumes for the collector
+  volumes: []
+  # - name: config-volume
+  #   configMap:
+  #     name: config
+
+  # Init containers for the collector
+  initContainers: []
+  # - name: init-nginx
+  #   image: nginx
+
+  # Additional containers for the collector
+  additionalContainers: []
+  # - name: additional-container
+  #   image: busybox
+
+  # Topology spread constraints for the collector
+  topologySpreadConstraints: []
+  # - maxSkew: 1
+  #   topologyKey: kubernetes.io/hostname
+  #   whenUnsatisfiable: DoNotSchedule
+  #   labelSelector:
+  #     matchLabels:
+  #       app: my-app
+
+  # Config maps for the collector
+  configmaps: []
+  # - name: config
+  #   mountPath: /etc/config
+
+# Top level field specifying collectors configuration
+collectors: {}
```

Comment:
Let's add a comment that helps users understand how they configure this field. [We do this in the demo chart](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/0d569572313bbbde68a09db945e9bbbfde16a582/charts/opentelemetry-demo/values.yaml#L46).  Also, if this is a list, the default should be `[]` I believe.

Reply:
i prefer a map here because it allows for more complex and specific overrides as helm doesn't allow for overriding or modifying array contacts. happy to explain more why this is done

Reply:
Is it a map where each key is a new collector? Let's make sure the values.yaml makes it easy to understand how to configure this section properly.

Reply:
yes, that's the idea. can do!

Reply:
the values.schema.json should also be type checking the contents for validity as well.